Gems/ROS2/Code/Source/RobotImporter/URDF/ArticulationsMaker.cpp

```diff
@@ -19,51 +20,51 @@ namespace ROS2
     namespace
     {
         using ArticulationCfg = PhysX::EditorArticulationLinkConfiguration;
-        static const AZStd::unordered_map<int, PhysX::ArticulationJointType> SupportedJointTypes{ {
-            { urdf::Joint::REVOLUTE, PhysX::ArticulationJointType::Hinge },
-            { urdf::Joint::CONTINUOUS, PhysX::ArticulationJointType::Hinge },
-            { urdf::Joint::PRISMATIC, PhysX::ArticulationJointType::Prismatic },
-            { urdf::Joint::FIXED, PhysX::ArticulationJointType::Fix },
+        static const AZStd::unordered_map<sdf::JointType, PhysX::ArticulationJointType> SupportedJointTypes{ {
+            { sdf::JointType::REVOLUTE, PhysX::ArticulationJointType::Hinge },
+            { sdf::JointType::CONTINUOUS, PhysX::ArticulationJointType::Hinge },
+            { sdf::JointType::PRISMATIC, PhysX::ArticulationJointType::Prismatic },
+            { sdf::JointType::FIXED, PhysX::ArticulationJointType::Fix },
         } };
     } // namespace
 
-    ArticulationCfg& AddToArticulationConfig(ArticulationCfg& articulationLinkConfiguration, const urdf::JointSharedPtr joint)
+    ArticulationCfg& AddToArticulationConfig(ArticulationCfg& articulationLinkConfiguration, const sdf::Joint* joint)
     {
         if (!joint)
         {
             return articulationLinkConfiguration;
         }
-        auto supportedArticulationType = SupportedJointTypes.find(joint->type);
+        auto supportedArticulationType = SupportedJointTypes.find(joint->Type());
         AZ_Warning(
             "ArticulationsMaker",
             supportedArticulationType != SupportedJointTypes.end(),
             "Articulations do not support type %d for URDF joint %s.",
-            joint->type,
-            joint->name.c_str());
+            static_cast<int>(joint->Type()),
+            joint->Name().c_str());
         if (supportedArticulationType != SupportedJointTypes.end())
         {
             const auto type = supportedArticulationType->second;
             articulationLinkConfiguration.m_articulationJointType = type;
             const AZ::Vector3 o3deJointDir{ 1.0, 0.0, 0.0 };
-            const AZ::Vector3 jointAxis = URDF::TypeConversions::ConvertVector3(joint->axis);
+            const AZ::Vector3 jointAxis = URDF::TypeConversions::ConvertVector3(joint->Axis()->Xyz());
```

Comment:
In case the joint has a fixed type, the `Axis` component [is removed](https://github.com/gazebosim/sdformat/blob/7f63d022a48d3a224a3b7a777316c3822525c7b2/src/parser_urdf.cc#L3153), which causes segfault here (`joint->Axis()` returns `nullptr`).